x-pack/plugins/secops/server/graphql/ecs/resolvers.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { GraphQLScalarType, Kind } from 'graphql';
+import { isBoolean, isNumber, isObject } from 'lodash/fp';
+
+/*
+ *  serialize: gets invoked when serializing the result to send it back to a client.
+ *
+ *  parseValue: gets invoked to parse client input that was passed through variables.
+ *
+ *  parseLiteral: gets invoked to parse client input that was passed inline in the query.
+ */
+
+export const toStringArrayScalar = new GraphQLScalarType({
+  name: 'DetailItemValue',
+  description: 'Represents value in detail item from the timeline who wants to more than one type',
+  serialize(value): string[] | null {
+    if (value == null) {
+      return null;
+    } else if (Array.isArray(value)) {
+      return convertArrayToString(value) as string[];
+    } else if (isBoolean(value) || isNumber(value) || isObject(value)) {
+      return [convertToString(value)];
+    }
+    return [value];
+  },
+  parseValue(value) {
+    return value;
+  },
+  parseLiteral(ast) {
+    switch (ast.kind) {
+      case Kind.INT:
+        return parseInt(ast.value, 10);
+      case Kind.FLOAT:
+        return Number(ast.value);
```

Comment:
We use `float` pretty much everywhere, but there are differences between `Number(value)` vs using `parseFloat(value)`

`Number('')` will return `0`, where `parseFloat('')` will return `NaN`

Also within the file:
x-pack/plugins/secops/server/graphql/events/resolvers.ts

you already use parseFloat:

```ts
      case Kind.FLOAT:
        return parseFloat(ast.value);
```

Can we collapse this into one re-usable function since they look like duplicates of each other?

Otherwise, I would suggest using `parseFloat(ast.value)` here as you have in graphql/events/resolvers.ts.

Reply:
I will do parseFloat but for now, I am not ready to merge it to one function yet since these two scalars since one is to match ES and the other one should give us more control.

Reply:
I am not ready to make a generic function for these two scalars since they are really different in theory. One is there to give back what ES has and the other one is there to give us more control.

I agree with you with parseFloat!!!